src/backend/expungeservice/request/request.py

```diff
@@ -0,0 +1,26 @@
+from flask import g
+import os
+from expungeservice.database import Database
+
+
+def before():
+
+    host = os.environ['PGHOST']
+    port = os.environ['PGPORT']
+    name = os.environ['PGDATABASE']
+    username = os.environ['PGUSER']
+    password = os.environ['POSTGRES_PASSWORD']
+
+    g.database = Database(
+        host=host, 
+        port=port, 
+        name=name, 
+        username=username, 
+        password=password)
+
+def teardown(exception):
+    try: 
+        g.database.close_connection()
+    except: pass
+
+    if exception: raise exception
```

Comment:
I'm not familiar with exceptions being passed to teardown, and what the flasky way is for dealing with them. Do you have a good reference I can read regarding this?

Reply:
Teardown functions in flask run regardless of whether a request succeeded, and may get passed an exception, whereas after_request functions only run if the request succeeded. So any necessary exception-handling belongs in teardown. 

I confess that I didn't look this through very carefully, but it looks like there is no need to re-raise errors that aren't handled in teardown. Unhandled errors are already raised to the server context. Some documentation: http://flask.pocoo.org/docs/0.12/reqcontext/#callbacks-and-errors

I'll remove the raise command here.

Reply:
In fact, the Database object @erikjasso wrote has a close_connection() function that it looks safe to just call without extra condition-checking.

Reply:
Thanks for explaining, and looking into it further